apps/src/sites/studio/pages/maker/discountcode.js

```diff
@@ -8,10 +8,27 @@ import getScriptData from '@cdo/apps/util/getScriptData';
 $(document).ready(() => {
   const scriptData = getScriptData('discountcode');
 
+  let eligibilityStatus = scriptData.eligibility_status;
+
+  // Generating users with the right eligibility status is non-trivial. As a way
+  // to get around this and be able to test this feature more easily, allow tester
+  // to put data in sessionStorage to override what the server says.
+  // We may want to remove this before shipping, though we'll likely want to make
+  // the server resilient to this anyways.
+  // To set, do something like:
+  // sessionStorage.setItem('testOnlyEligibilityStatus', JSON.stringify({is_pd_eligible: true, is_progress_eligible: true}))
+  const testOnlyEligibilityStatus = sessionStorage.getItem('testOnlyEligibilityStatus');
```

Comment:
This seems like a thing we should remove now, even if we _are_ going to make the server resilient to it later.  It's not used in any automated testing, so from a production point of view it's effectively dead code.  Also, if all this does is adjust the props we pass to `<EligibilityChecklist>`, would storybook be a good enough approach for local testing?

Reply:
The server is currently resilient to it, given that we don't post anything.  

My thoughts were were that storybook is great at individual controls, but for an entire page, getting all the formatting right often requires you to look at this outside of storybook. The scenarios for all of our different states can be difficult to construct, and this seemed like an easier way to accomplish this (especially given that we don't actually have all the server logic built yet).

Reply:
That's a good argument.  :+1: Leave it in.